extensions/blocks/anchor-fm/anchor-fm.php

```diff
@@ -0,0 +1,109 @@
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Podcast_Helper;
+
+const FEATURE_NAME = 'anchor-fm';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers Anchor.fm integration for the block editor.
+ */
+function register_block() {
+	Blocks::jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'plan_check' => false,
+		)
+	);
+
+	// Register post_meta for connecting Anchor podcasts with posts.
+	register_post_meta(
+		'post',
+		'anchor_podcast',
+		array(
+			'show_in_rest' => true,
+			'single'       => true,
+			'type'         => 'string',
+		)
+	);
+	register_post_meta(
+		'post',
+		'anchor_episode',
+		array(
+			'show_in_rest' => true,
+			'single'       => true,
+			'type'         => 'string',
+		)
+	);
+	register_post_meta(
+		'post',
+		'anchor_spotify_show',
+		array(
+			'show_in_rest' => true,
+			'single'       => true,
+			'type'         => 'string',
+		)
+	);
+}
+
+/**
+ * Checks URL params to determine the Anchor integration action to perform.
+ */
+function process_anchor_params() {
+	$current_screen = \get_current_screen();
+	// TODO: Replace `$current_screen->is_block_editor()` with `wp_should_load_block_editor_scripts_and_styles()` that is introduced in WP 5.6.
+	if ( method_exists( $current_screen, 'is_block_editor' ) && ! $current_screen->is_block_editor() ) {
+		// Return early if we are not in the block editor.
+		return;
+	}
+
+	// phpcs:disable WordPress.Security.NonceVerification.Recommended
+	$podcast_id       = isset( $_GET['anchor_podcast'] ) ? $_GET['anchor_podcast'] : null;
+	$episode_id       = isset( $_GET['anchor_episode'] ) ? $_GET['anchor_episode'] : null;
+	$spotify_show_url = isset( $_GET['spotify_show_url'] ) ? $_GET['spotify_show_url'] : null;
+	// phpcs:enable WordPress.Security.NonceVerification.Recommended
+
+	$data    = array();
+	$post_id = get_post()->ID;
+
+	if ( ! empty( $podcast_id ) ) {
+		$feed = 'https://anchor.fm/s/' . $podcast_id . '/podcast/rss';
+		$rss  = Jetpack_Podcast_Helper::load_feed( $feed );
+		if ( ! \is_wp_error( $rss ) ) {
+			$data['podcastId'] = $podcast_id;
+			update_post_meta( $post_id, 'anchor_podcast', $podcast_id );
+
+			if ( ! empty( $episode_id ) ) {
+				$track = Jetpack_Podcast_Helper::get_track_data( $feed, $episode_id );
+				if ( ! \is_wp_error( $track ) ) {
+					$data['episodeId'] = $episode_id;
+					$data['track']     = $track;
+					update_post_meta( $post_id, 'anchor_episode', $episode_id );
+				}
+			}
+		}
+	}
+
+	if ( ! empty( $spotify_show_url ) ) {
+		$data['spotifyShowUrl'] = $spotify_show_url;
+		if ( get_post_meta( $post_id, 'anchor_spotify_show', true ) !== $spotify_show_url ) {
+			update_post_meta( $post_id, 'anchor_spotify_show', $spotify_show_url );
+			$data['action'] = 'insert-spotify-badge';
+		}
+	}
+
+	wp_localize_script( 'jetpack-blocks-editor', 'Jetpack_AnchorFm', $data );
+}
+
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+add_action( 'enqueue_block_assets', __NAMESPACE__ . '\process_anchor_params' );
```

Comment:
Does this mean that it will get added all the time, even when not dealing with this block?

Reply:
Correct, but this is because this is not actually an block but an extension that performs some logic based on the URL params.

Reply:
I think we should move this kind of implementation to another location since as you said it isn't a block.
we've discussed this issue before. Putting it in the `/extensions/blocks` get confusion.
Maybe something like `extensions/features` or something like that

Reply:
@retrofox moved the `extensions/features` per suggestion

Reply:
> Maybe something like extensions/features or something like that

I think that needs a big refactor, since right now only the `blocks` folder is used for building/loading extensions:

https://github.com/Automattic/jetpack/blob/c733efd860f5c863fd8c9e307978c6d34680d345/class.jetpack-gutenberg.php#L327-L341

That will also need to change the tooling in the WP.com side, since right now Phabricator only generates builds from the source living in the `blocks` dir.

We cannot simply move the code to a different location, we need to change the whole building process which can be considered a giant task, so I'll keep the extension in the `blocks` folder.

Reply:
I think `blocks` is okay here for now. In the future though, I agree that it would be nice to keep making some order in our blocks and extensions, to have a clear differenciation for each. It has become very hard for new contributors to figure out where their code should leave, when we have some block extensions like External Media being `extensions/shared/` components, others like Social Previews or the Video block extension being in `extensions/blocks`, and wpcom paid block extensions in `extensions/paid-blocks`.

It should probably be its own project though.

Reply:
> I think blocks is okay here for now. In the future though, I agree that it would be nice to keep making some order in our blocks and extensions, to have a clear differenciation for each. It has become very hard for new contributors to figure out where their code should leave

Yeah, we talked about this when we were working on `Social Preview`? when at that moment `future` was `now`? 😅  Clearly, it isn't a block but there it is.

>  and wpcom paid block extensions in extensions/paid-blocks.

Actually, `paid-blocks` is in `extensions/extended-blocks` folder. This is a good example of what we had to do to move some implementations that didn't fit very well in the `extensions/blocks` folder, even being them `blocks`.

<img src="https://user-images.githubusercontent.com/77539/101622896-67bb5380-39f6-11eb-941e-a10e95fd9385.png" width="300px" />


> I think that needs a big refactor, since right now only the blocks folder is used for building/loading extensions:

I'm not sure how big it could be. It's something that we did before, in the middle of the premium block feature.

> We cannot simply move the code to a different location, we need to change the whole building process which can be considered a giant task, so I'll keep the extension in the blocks folder.

Said that way, _giant task_ sounds almost impossible. I disagree.

> ...so I'll keep the extension in the blocks folder.

Agree with this, too 🚀 if it makes the implementation goes faster, nice! It's your call.